core/domain/image_services.py

```diff
@@ -0,0 +1,103 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Service for managing operations on images."""
+
+from core.domain import fs_domain
+from core.platform import models
+import feconf
+
+gae_image_services = models.Registry.import_gae_image_services()
+
+
+def get_image_dimensions(filename, exp_id):
+    """Gets the dimensions of the image with the given filename
+
+    Args:
+        filename: str. The filename of the image in exploration.
+        exp_id: str. ID of the exploration.
+
+    Returns:
+        tuple. Returns height and width of the image.
+    """
+    # Because the filepath in DEV_MODE is 'filename' whereas in PROD_MODE
+    # it is 'image/filename'.
+    filepath = (
+        filename if (
+            feconf.DEV_MODE)
+        else 'image/%s' % filename)
+
+    file_system_class = (
+        fs_domain.ExplorationFileSystem if (
+            feconf.DEV_MODE)
+        else fs_domain.GcsFileSystem)
+    fs = fs_domain.AbstractFileSystem(file_system_class(exp_id))
+    content = fs.get(filepath)
+    height, width = gae_image_services.get_image_dimensions(content)
+    return height, width
+
+
+def create_compressed_versions_of_image(filename, exp_id):
```

Comment:
This image_services module should deal only with images, not with the file system. The function args should be (image_content, compression_factor).

Reply:
That said, if you drop the above function, then maybe this last function should go into exp_services and be called exp_services.save_image_file(exp_id, filename). Then you can leave it mostly as-is.

The reason for the difference is that a file called "image_services" should be dealing with images, whereas a file called "exp_services" should be dealing with all components of an exploration. Given what you're doing here, this probably fits more naturally in the latter than the former.

Reply:
That sounds good. So I'll delete the core/domain/image_services.py and add add these functions to the exp_services.py which will handle the file system as well as the calls to the image service.
I think the name should still be create_compressed_versions_of_the_image.  The reason behind this is that it is suitable for both the one off job and the call from editor.py while saving a new image.
Plus the name gives a clear description of what it does.

Reply:
Yeah. Done